sdk/Trace/BatchSpanProcessor.php

```diff
@@ -77,11 +82,23 @@ public function onStart(API\Span $span): void
      */
     public function onEnd(API\Span $span): void
     {
-        if (count($this->queue) < $this->maxQueueSize) {
-            $this->queue[] = $span;
+        if (null !== $this->exporter && $this->running) {
+            if (count($this->queue) < $this->maxQueueSize) {
+                $this->queue[] = $span;
+            }
+
+            if ($this->bufferReachedExportLimit() || $this->enoughTimeHasPassed()) {
```

Comment:
The spec is very vague on the interplay between `maxExportBatchSize` and `scheduledDelayMillis` but both the python and go SDK implementations appear to send a batch for whichever occurs first, not necessarily waiting for both conditions to be met. This change brings the php sdk in line with the other implementations.

Reply:
I think that the `whichever occurs first` scenario seems to make sense to me.  We should send a batch when we run out of space or time :+1: